src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/HtmlCSharp/CompletionHandler.cs

```diff
@@ -123,6 +123,28 @@ internal class CompletionHandler : IRequestHandler<CompletionParams, SumType<Com
                 result = SetResolveData(result.Value, serverKind);
             }
 
+            // This is a bit fraught. Basically in the scenario That someone types "@<slight pause>{" they'll get @__builder because '{' is a commit character.
+            // Our solution is to make the results in this case empty, which is a bit of a lie.
+            // It's the best we can do for now, if a user complains we can revisit.
+            if (result.HasValue && request.Context.TriggerCharacter != null && request.Context.TriggerCharacter.Equals("@"))
+            {
+                if (result.Value.Value is CompletionItem[] items)
+                {
+                    foreach (var item in items)
+                    {
+                        item.InsertText = string.Empty;
+                    }
+                }
+                else
+                {
+                    var list = result.Value.Value as CompletionList;
+                    foreach (var item in list?.Items)
+                    {
+                        item.InsertText = string.Empty;
+                    }
+                }
+            }
```

Comment:
var items = result.Value.Match(
                    itemArray => itemArray,
                    list => list?.Items);

                foreach (var item in items)
                {
                    item.InsertText = string.Empty;
                }

Reply:
Any reason not to use the Match method?

Reply:
Frankly I misunderstood its purpose, but seeing it laid out here your way is clearly better.